airbyte-integrations/connectors/source-chargebee/source_chargebee/streams.py

```diff
@@ -132,6 +117,77 @@ def read_records(
         # Always return an empty generator just in case no records were ever yielded
         yield from []
 
+
+class SemiIncrementalChargebeeStream(ChargebeeStream):
```

Comment:
can you add a docstring explaining what a "semi incremental" stream is? maybe a better name is "ClientSideIncrementalStream" ? Or potentially a better structure would be: 

1. BaseIncrementalStream <-- contains all common functionality for incremental
2. ClientSideIncrementalStream <-- does a full refresh read but filters on the client side
3. ServerSideIncrementalStream <-- does a proper incremental read

Reply:
Added docstring.

Reply:
btw, if I'll add `BaseIncrementalStream` it will contain only `__init__` function because all other function will have different implementation depending on which class it is: `ClientSideIncrementalStream` or `ServerSideIncrementalStream`.